### DISCUSSION TITLE
typescript multiple selector state

### DISCUSSION THREAD

**Discussion by callmejm** [pmndrs/zustand#1824] [General] [answered]
In subscribe , can I only select `bears` and `trees` state ?

```ts
//store.ts
import { create } from 'zustand'
import { subscribeWithSelector } from 'zustand/middleware'
interface BearState {
  bears: number,
  trees: number,
  clouds: number,
  increase: (by: number) => void
}

export const useBearStore = create<BearState>()(subscribeWithSelector<BearState>((set,get) => ({
  bears: 0,
  trees: 0,
  clouds: 0,
  increase: (by) => set((state) => ({ bears: state.bears + by })),
})))

```

```ts
//App.tsx
function App() {
    const bearStore = useBearStore()
    useEffect(() => {
        const unsubscribe = useBearStore.subscribe(
          (state) => [state.bears, state.trees],
          (current, previous) => {
            //selected state
            console.log(current.bears)
             console.log(current.trees)
          },
        );
    
        return () => {
          unsubscribe();
        }
      }, []);
    return (
        <div></div>
    )
}

export default App
```

**Comment by dbritto-dev** [comment]
@callmejm yes, you can

```ts
const { bears, trees } = useBearStore(state => ({ bears: state.bears, trees: stare.trees }))
```

**Reply by callmejm** [reply]
what I mean is happen in subscribe , so that I using `subscribeWithSelector`

**Reply by dbritto-dev** [reply]
> what I mean is happen in subscribe , so that I using `subscribeWithSelector`

Yeah, that's how works

**Reply by callmejm** [reply]
sorry maybe I don't get the meaning, base on your suggestion code , how it related to my App.tsx `useBearStore.subscribe` ? `(current, previous)` not return correct value

**Reply by callmejm** [reply]
What I want is when `bears` and `trees` get update only get the callback from subscribe , if `clouds` is update , I don't want subscribe trigger callback

**Reply by dbritto-dev** [reply]
@callmejm that's how works subscribe with selector. https://github.com/pmndrs/zustand#using-subscribe-with-selector

**Reply by callmejm** [reply]
doc is not clear enough, so that I get confuse, why my code doesn't work

**Reply by dbritto-dev** [reply]
> doc is not clear enough, so that I get confuse, why my code doesn't work

Would you mind telling me which part is not clear enough?

**Reply by callmejm** [reply]
sorry that, I think maybe I ask again: how can I get `current.bears` inside my subscribe ? because `current` now is an array

**Reply by dbritto-dev** [reply, marked as answer]
@callmejm here you go https://codesandbox.io/p/sandbox/goofy-burnell-vop1fm?file=%2Fsrc%2FApp.tsx%3A22%2C3-36%2C10

**Solution**

Return an object instead of an array

```ts
(state) => ({ bears: state.bears, trees: state.trees }),
```

**Reply by callmejm** [reply]
thank you
